import-service/src/handlers/importFileParser.ts

```diff
@@ -0,0 +1,52 @@
+import { S3Handler, S3Event } from 'aws-lambda';
+import { S3 } from 'aws-sdk';
+import { BUCKET, REGION, UPLOADED_FILES_DIR, PARSED_FILES_DIR } from '../constants';
+import { parseCSVRecord } from '../utils';
+
+export const importFileParser: S3Handler = async (
+    event: S3Event,
+): Promise<void> => {
+    console.log('Lambda invocation with event: ', JSON.stringify(event, null, 4));
+
+    try {
+        const { Records } = event;
+        const s3 = new S3({ region: REGION });
+
+        for (const record of Records) {
+            const { key: recordKey } = record.s3.object;
+            console.log('Using bucket: ', BUCKET);
+            const params = {
+                Bucket: BUCKET,
+                Key: recordKey,
+            };
+
+            console.log('Get record object from storage: ', recordKey);
+            const s3Object = s3.getObject(params);
+
+            console.log('Parse record data..');
+            await parseCSVRecord(s3Object);
+            console.log('Record data successfully parsed!');
+
+            console.log('Start copying from: ', recordKey);
+            const newRecordKey = recordKey.replace(UPLOADED_FILES_DIR, PARSED_FILES_DIR);
+            await s3
+                .copyObject({
+                    Bucket: BUCKET,
+                    Key: newRecordKey,
+                    CopySource: `${BUCKET}/${recordKey}`,
+                })
+                .promise();
+            console.log('Record successfully copied into: ', newRecordKey);
+
+            console.log('Deleting old record version from: ', recordKey);
+            await s3.deleteObject({ Bucket: BUCKET, Key: recordKey }).promise();
+            console.log('Old record version successfully deleted!');
+        }
+
+        console.log('File processed successfully!');
+    } catch (error) {
+        console.error('Error: ', error);
+    }
+
+    return;
```

Comment:
Hi! Everything is good, though there is a note in task description saying that in `importFileParser` _"The response should be a correct HTTP response code."_, but you're returning an undefined value here. I'm not certain about importance of that moment and it's not gonna affect the score, nonetheless it's kind of a mistake.

Reply:
Hi!
I was confused by this moment in the description of the task, because `S3Handler` demanded to return `void`.
I can just extend this interface, but I agree with you that the moment is not that important.